MarsRover/Headings/EastHeading.cs

```diff
@@ -0,0 +1,27 @@
+﻿
+namespace MarsRover
+{
+    public class EastHeading : IHeading
+    {
+        public IHeading TurnRight() => new SouthHeading();
+
+        public IHeading TurnLeft() => new NorthHeading();
+
+
+        public Grid MoveForward(Grid grid)
+        {
+            var x = grid.XCoordinate(); 
```

Comment:
Looking at these movements, I think these should return `void`.  Seeing this `new`-up an object every time it moves is a bit concerning...why not just update the existing object instead of creating a new one each time?  We should try to minimize the number of objects we create.

I imagine this function should look something like (notice it takes a `Rover` instead of a `Grid`):

```cs
public void MoveForward(Rover rover)
{
    var currentX = rover.XCoordinate();
    var newX = (rover.XCoordinate() < rover.GridWidth()) ? currentX += 1 : 0;
    rover.SetX(newX);
}
```

A grid should be a static thing...once it's created, there shouldn't be a need to update it unless we had advance functionality like dynamic terrain or something crazy.  It's the rover itself that changes location, so lets update the rover instead.

(this applies to other similar classes and functions, like `MoveBackward` below)

Reply:
I updated this as you suggested with void functions and used the `Rover` class to set the coordinates.  I still do need the size of the grid to do my calculations so you will see that passed in as well.